avro/src/main/java/com/fasterxml/jackson/dataformat/avro/deser/JacksonAvroParserImpl.java

```diff
@@ -700,6 +700,12 @@ private final void _finishLongText(int len) throws IOException
                 break;
             case 3: // 4-byte UTF
                 c = _decodeUTF8_4(c);
+                // if outPtr >= outBuf.length before adding
+                // the input is malformed.
```

Comment:
Actually, not necessarily so. But we do need to check there's room. I'll tweak this PR a bit.